invenio_stats/tasks.py

```diff
@@ -123,36 +116,40 @@ def _success_date():
              _success_date(),
              stats_only=True)
 
-    def date_range(self, start_date, end_date):
-        """Get all dates that haven't been processed."""
-        if start_date >= end_date:
-            for n in range((start_date - end_date).days + 1):
-                yield end_date + datetime.timedelta(n)
-        else:
-            for n in range((end_date - start_date).days + 1):
-                yield start_date + datetime.timedelta(n)
-
-    def agg_iter(self, date):
+    def agg_iter(self):
         """Aggregate and return dictionary to be indexed in ES."""
         aggregation_data = {}
-        result = self.event_agg_query.execute()
-        for aggregation in result.aggregations:
-            for hit in result.aggregations[aggregation]['buckets']:
-                aggregation_data[self.aggregation_field] = hit['key']
-                aggregation_data['count'] = hit['doc_count']
-                print('aggregation_data:', aggregation_data)
-
-                tobeindexed = dict(_id='{0}_{1}'.
-                                   format(hit['key'],
-                                          date),
-                                   _index='stats-{0}-{1}'.
-                                   format(self.event,
-                                          date.strftime(
-                                              self.index_name_suffix)),
-                                   _type='{}-aggregation'.
-                                   format(self.event),
-                                   _source=aggregation_data)
-                yield tobeindexed
+        self.agg_query = Search(using=self.client,
+                                index='events-stats-file-download').\
+            filter('range', timestamp={'gte': self.get_bookmark().isoformat(),
+                                       'lt': 'now'})
+        self.agg_query.aggs.bucket('per_{}'.format(self.aggregation_interval),
+                                   'date_histogram',
+                                   field='timestamp',
+                                   interval=self.aggregation_interval)
```

Comment:
How does this scale with many files? The result would be huge if you ask a year with `aggregation_interval='day'`